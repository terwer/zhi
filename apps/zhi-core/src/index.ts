/*
 * Copyright (c) 2023, Terwer . All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Terwer designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Terwer in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Terwer, Shenzhen, Guangdong, China, youweics@163.com
 * or visit www.terwer.space if you need additional information or have any
 * questions.
 */

/**
 * @packageDocumentation
 * zhi-core 主题核心模块
 */

import Zhi from "./theme/zhi"
import DeviceDetection from "zhi-device-detection"

// 主题样式注入入口
import "./style/common/fonts/webfont.css"
import "./style/index.styl"

const loadTheme = async (): Promise<void> => {
  const zhi = new Zhi(DeviceDetection.getDevice())
  await zhi.init()
}

/**
 * 主题入口，由思源笔记自动触发，请勿主动调用
 */
;(async () => {
  await loadTheme()
})()

export default loadTheme
export { Zhi }
