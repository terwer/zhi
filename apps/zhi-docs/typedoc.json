{
  "plugin": [
    "typedoc-plugin-frontmatter",
    "./node_modules/typedoc-plugin-markdown/dist",
    "./node_modules/typedoc-vitepress-theme/dist"
  ],
  "entryPoints": [
    "../../libs/zhi-device/src/index.ts",
    "../../libs/zhi-env/src/index.ts",
    "../../libs/zhi-log/src/index.ts",
    "../../libs/zhi-common/src/index.ts",
    "../../libs/zhi-blog-api/src/index.ts",
    "../../libs/zhi-fetch-middleware/src/index.ts",
    "../../libs/zhi-xmlrpc-middleware/src/index.ts"
  ],
  "tsconfig": "./tsconfig.json",
  "frontmatterTags": ["author", "description", "outline"],
  "frontmatterGlobals": {
    "outline": "deep"
  }
}