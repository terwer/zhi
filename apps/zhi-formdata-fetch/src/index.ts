import FormdataFetch from "./lib/formdataFetch"
import fetch, { Blob, FormData, Headers } from "node-fetch"

export { FormdataFetch }
export { fetch }
export { Blob, FormData, Headers }
