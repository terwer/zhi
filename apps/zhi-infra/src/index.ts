import ZhiInfra from "./zhiInfra"
import "./lib/requireHacker"
import { SiyuanDevice } from "zhi-device"
import { ILogger, MainFunction, safeParseArgs, simpleLogger } from "zhi-lib-base"
import { CustomCmd } from "zhi-cmd"

/**
 * 基础设施初始化入口
 *
 * @param args
 *   zhiNpmPath - 内置的 NPM 目录
 *   isFixPath - 是否修复路径
 */
const main: MainFunction = async (args?: any[]): Promise<void> => {
  const logger: ILogger = simpleLogger("init-infra", "zhi", false)

  // win
  const win = SiyuanDevice.siyuanWindow()
  win.zhi = win.zhi ?? {}
  win.zhi.status = win.zhi.status ?? {}

  // mountDevice
  if (!win.zhi.status.deviceInited) {
    win.zhi.device = SiyuanDevice
    win.zhi.status.deviceInited = true
    logger.info("zhi device inited")
  } else {
    logger.info("zhi device is already inited.skip")
  }

  // mountCmd
  if (!win.zhi.status.cmdInited) {
    const cmd = new CustomCmd()
    win.zhi.cmd = cmd
    win.zhi.status.cmdInited = true
    logger.info("zhi cmd inited")
  } else {
    logger.info("zhi cmd is already inited.skip")
  }

  // mountNpmManager
  const depsJsonPath: string = safeParseArgs(args, 0)
  const isFixPath: boolean = safeParseArgs(args, 1)
  if (!win.zhi.status.infraInited) {
    const infra = new ZhiInfra(depsJsonPath)
    if (isFixPath) {
      infra.fixPathEnv()
    }
    await infra.hackRequire()
    win.zhi.npm = infra.getNpmManager()
    win.zhi.status.infraInited = true
    logger.info("zhi infra inited")
  } else {
    logger.info("zhi infra is already inited.skip")
  }
}

export default main
