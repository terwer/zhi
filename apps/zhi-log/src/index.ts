/**
 * @packageDocumentation
 * zhi-log 多平台日志模块
 */

import Env from "zhi-env"
import LogLevelEnum, { LogConstants } from "./lib/logConstants"
import AbstractLogFactory from "./lib/factory/abstractLogFactory"
import CustomLogFactory from "./lib/factory/customLogFactory"
import DefaultLogger from "./lib/defaultLogger"
import EnvHelper from "./lib/envHelper"
import crossChalk from "./lib/crossChalk"

/**
 * 日志工具类
 *
 * @public
 * @author terwer
 * @since 1.0.7
 */
class LogFactory {
  /**
   * 默认日志记录器
   *
   * @param stackSize - 栈的深度
   * @param env - 环境变量实例
   */
  public static defaultLogger(env?: Env, stackSize?: number): DefaultLogger {
    return LogFactory.customLogFactory(undefined, undefined, env).getLogger(undefined, stackSize)
  }

  /**
   * 自定义日志工厂
   *
   * @param level - 级别
   * @param sign - 标志
   * @param env - 环境变量
   */
  public static customLogFactory(level?: LogLevelEnum, sign?: string, env?: Env) {
    return new CustomLogFactory(level, sign, env)
  }

  /**
   * 自定义日志工厂，自定义前缀
   *
   * @param sign - 标志
   * @param env - 环境变量
   */
  public static customSignLogFactory(sign?: string, env?: Env) {
    return new CustomLogFactory(undefined, sign, env)
  }
}
export default LogFactory

export { LogLevelEnum, AbstractLogFactory, CustomLogFactory }
export { LogConstants, EnvHelper }
export { crossChalk }
export type { DefaultLogger }
