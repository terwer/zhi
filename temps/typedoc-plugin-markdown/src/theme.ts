import * as prettier from 'prettier';
import {
  PageEvent,
  ProjectReflection,
  Reflection,
  RenderTemplate,
  Renderer,
  Theme,
} from 'typedoc';
import { NavigationBuilder } from './converter/navigation-builder';
import { UrlBuilder } from './converter/url-builder';
import { MarkdownThemeRenderContext } from './theme-render-context';

export class MarkdownTheme extends Theme {
  private _renderContext: MarkdownThemeRenderContext;
  private _prettierOptions: prettier.Options | null;

  constructor(renderer: Renderer) {
    super(renderer);

    // Warnings
    if (this.application.options.getValue('enableFrontmatter')) {
      this.application.logger.warn(
        '[typedoc-plugin-markdown] `enableFrontmatter` has been removed. Please use plugin-typedoc-frontmatter instead',
      );
    }

    this.listenTo(this.owner, {
      [PageEvent.BEGIN]: this.onBeginPage,
    });
  }

  getRenderContext() {
    if (!this._renderContext) {
      this._renderContext = new MarkdownThemeRenderContext(
        this,
        this.application.options,
      );
    }
    return this._renderContext;
  }

  getPrettierOptions() {
    if (!this._prettierOptions) {
      this._prettierOptions = this.resolvePrettierOptions();
    }
    return this._prettierOptions;
  }

  resolvePrettierOptions() {
    const prettierFile = prettier.resolveConfigFile.sync();
    return prettierFile ? prettier.resolveConfig.sync(prettierFile) : null;
  }

  render(
    page: PageEvent<Reflection>,
    template: RenderTemplate<PageEvent<Reflection>>,
  ): string {
    return prettier.format(template(page) as string, {
      ...this.getPrettierOptions(),
      parser: 'markdown',
    });
  }

  getUrls(project: ProjectReflection) {
    const urls = new UrlBuilder(this.getRenderContext()).getUrls(project);
    return urls;
  }

  getNavigation(project: ProjectReflection) {
    const navigation = new NavigationBuilder(
      this.getRenderContext(),
    ).getNavigation(project);
    return navigation;
  }

  protected onBeginPage(page: PageEvent) {
    this.getRenderContext().activeLocation = page.url;
  }
}
