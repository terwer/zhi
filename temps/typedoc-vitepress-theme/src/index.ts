import * as fs from 'fs';
import * as path from 'path';
import { Application, ParameterType } from 'typedoc';
import {
  MarkdownPluginOptionsReader,
  MarkdownRendererEvent,
} from 'typedoc-plugin-markdown';
import { SidebarOptions } from './model';
import { DEFAULT_SIDEBAR_OPTIONS } from './options';
import { getSidebar } from './sidebars/sidebars';

export function load(app: Application) {
  app.options.addDeclaration({
    name: 'docsRoot',
    help: '',
    type: ParameterType.Path,
    defaultValue: './docs',
  });

  app.options.addDeclaration({
    name: 'sidebar',
    help: '',
    type: ParameterType.Mixed,
    defaultValue: DEFAULT_SIDEBAR_OPTIONS,
  });

  app.options.addReader(
    new MarkdownPluginOptionsReader({
      anchorFormat: 'slug',
      entryDocument: 'index.md',
      hideBreadcrumbs: true,
      hidePageHeader: true,
      out: './docs/api',
    }),
  );

  app.renderer.postRenderAsyncJobs.push(
    async (output: MarkdownRendererEvent) => {
      const sidebarOptions = {
        ...DEFAULT_SIDEBAR_OPTIONS,
        ...(app.options.getValue('sidebar') as SidebarOptions),
      };
      if (sidebarOptions.autoConfiguration) {
        const outDir = app.options.getValue('out') as string;
        const sourceDir = app.options.getValue('docsRoot') as string;
        const sidebarPath = path.resolve(outDir, 'typedoc-sidebar.json');
        const basePath = path.relative(sourceDir, outDir);
        const sidebarJson = getSidebar(
          output.navigation,
          basePath,
          sidebarOptions,
        );
        fs.writeFileSync(sidebarPath, JSON.stringify(sidebarJson));
      }
    },
  );
}
