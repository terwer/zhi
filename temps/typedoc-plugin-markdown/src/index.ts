import { Application } from 'typedoc';
import { declareOptions } from './options/options';
import { generateMarkdown, renderMarkdown } from './renderer/renderer';
import { MarkdownTheme } from './theme';

export function load(app: Application) {
  /**
   * Exposes markdown and bootstrap the markdown theme to the renderer
   */
  app.renderer.defineTheme('markdown', MarkdownTheme);

  /**
   * Defines all plugin options
   */
  declareOptions(app);

  /**
   * Decouple HTML logic from the renderer (there should probably be a better solution to this)
   */
  Object.defineProperty(app, 'generateDocs', { value: generateMarkdown });
  Object.defineProperty(app.renderer, 'render', {
    value: renderMarkdown,
    configurable: true,
  });
}

/**
 * Expose global entrypoints
 */
export * from './models';
export * from './options/options-reader';
export { MarkdownRendererEvent } from './renderer/renderer';
export { partials } from './resources/resources';
export { MarkdownTheme } from './theme';
export { MarkdownThemeRenderContext } from './theme-render-context';
