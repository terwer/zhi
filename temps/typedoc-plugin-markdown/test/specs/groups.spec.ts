import { ProjectReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../src/theme-render-context';

describe(`Groups:`, () => {
  let project: ProjectReflection;
  let context: MarkdownThemeRenderContext;

  const baseOptions = {
    hideKindTag: true,
    disableSources: true,
    outputFileStrategy: 'modules',
  };

  const stubbedPartials = [
    'declarationMember',
    'reflectionMember',
    'signatureMember',
  ];

  describe(`(excludeGroups=false)`, () => {
    beforeAll(async () => {
      ({ project, context } = await global.bootstrap(['groups.ts'], {
        options: {
          ...baseOptions,
          excludeGroups: false,
        },
        stubPartials: stubbedPartials,
      }));
    });
    test(`should render group members'`, () => {
      expect(context.partials.members(project, 2)).toMatchSnapshot();
    });

    test(`should render group TOC'`, () => {
      expect(context.partials.toc(project)).toMatchSnapshot();
    });
  });

  describe(`(excludeGroups=true)`, () => {
    beforeAll(async () => {
      ({ project, context } = await global.bootstrap(['groups.ts'], {
        options: {
          ...baseOptions,
          excludeGroups: true,
        },
        stubPartials: stubbedPartials,
      }));
    });
    test(`should render group members'`, () => {
      expect(context.partials.members(project, 2)).toMatchSnapshot();
    });

    test(`should render group TOC'`, () => {
      expect(context.partials.toc(project)).toMatchSnapshot();
    });
  });
});
