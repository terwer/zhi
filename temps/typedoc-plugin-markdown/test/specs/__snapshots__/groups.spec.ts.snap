// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups: (excludeGroups=false) should render group TOC' 1`] = `
"## Index

### Enumerations

- [EnumA](exports.md#enuma)
- [EnumB](exports.md#enumb)

### Classes

- [ClassA](exports.md#classa)
- [ClassB](exports.md#classb)

### Interfaces

- [InterfaceA](exports.md#interfacea)
- [InterfaceB](exports.md#interfaceb)

### Type Aliases

- [typeA](exports.md#typea)
- [typeB](exports.md#typeb)

### Variables

- [variableA](exports.md#variablea)
- [variableB](exports.md#variableb)

### Functions

- [functionA](exports.md#functiona)
- [functionB](exports.md#functionb)
"
`;

exports[`Groups: (excludeGroups=false) should render group members' 1`] = `
"## Enumerations

### EnumA

[partial: reflectionMember]



---

### EnumB

[partial: reflectionMember]



---

## Classes

### ClassA

[partial: reflectionMember]



---

### ClassB

[partial: reflectionMember]



---

## Interfaces

### InterfaceA

[partial: reflectionMember]



---

### InterfaceB

[partial: reflectionMember]



---

## Type Aliases

### typeA

[partial: declarationMember]



---

### typeB

[partial: declarationMember]



---

## Variables

### variableA

[partial: declarationMember]



---

### variableB

[partial: declarationMember]



---

## Functions

### functionA()

[partial: signatureMember]



---

### functionB()

[partial: signatureMember]



---"
`;

exports[`Groups: (excludeGroups=true) should render group TOC' 1`] = `
"## Index

- [EnumA](exports.md#enuma)
- [EnumB](exports.md#enumb)
- [ClassA](exports.md#classa)
- [ClassB](exports.md#classb)
- [InterfaceA](exports.md#interfacea)
- [InterfaceB](exports.md#interfaceb)
- [typeA](exports.md#typea)
- [typeB](exports.md#typeb)
- [variableA](exports.md#variablea)
- [variableB](exports.md#variableb)
- [functionA](exports.md#functiona)
- [functionB](exports.md#functionb)"
`;

exports[`Groups: (excludeGroups=true) should render group members' 1`] = `
"## Enumerations

### EnumA

[partial: reflectionMember]



---

### EnumB

[partial: reflectionMember]



---

## Classes

### ClassA

[partial: reflectionMember]



---

### ClassB

[partial: reflectionMember]



---

## Interfaces

### InterfaceA

[partial: reflectionMember]



---

### InterfaceB

[partial: reflectionMember]



---

## Type Aliases

### typeA

[partial: declarationMember]



---

### typeB

[partial: declarationMember]



---

## Variables

### variableA

[partial: declarationMember]



---

### variableB

[partial: declarationMember]



---

## Functions

### functionA()

[partial: signatureMember]



---

### functionB()

[partial: signatureMember]



---"
`;
